src/AzureAppConfigurationImpl.ts

```diff
@@ -613,6 +637,118 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         }
         return featureFlagReference;
     }
+
+    async #generateAllocationId(featureFlag: any): Promise<string> {
+        let rawAllocationId = "";
+        // Only default variant when enabled and variants allocated by percentile involve in the experimentation
+        // The allocation id is genearted from default variant when enabled and percentile allocation
+        const variantsForExperiementation: string[] = [];
+
+        if (featureFlag[ALLOCATION_KEY_NAME]) {
+            rawAllocationId += `seed=${featureFlag[ALLOCATION_KEY_NAME][SEED_KEY_NAME] ?? ""}\ndefault_when_enabled=`;
+
+            if (featureFlag[ALLOCATION_KEY_NAME][DEFAULT_WHEN_ENABLED_KEY_NAME]) {
+                variantsForExperiementation.push(featureFlag[ALLOCATION_KEY_NAME][DEFAULT_WHEN_ENABLED_KEY_NAME]);
+                rawAllocationId += `${featureFlag[ALLOCATION_KEY_NAME][DEFAULT_WHEN_ENABLED_KEY_NAME]}`;
+            }
+
+            rawAllocationId += "\npercentiles=";
+
+            const percentileList = featureFlag[ALLOCATION_KEY_NAME][PERCENTILE_KEY_NAME];
+            if (percentileList) {
+                const sortedPercentileList = percentileList
+                    .filter(p =>
+                        (p[FROM_KEY_NAME] !== undefined) &&
+                        (p[TO_KEY_NAME] !== undefined) &&
+                        (p[VARIANT_KEY_NAME] !== undefined) &&
+                        (p[FROM_KEY_NAME] !== p[TO_KEY_NAME]))
+                    .sort((a, b) => a[FROM_KEY_NAME] - b[FROM_KEY_NAME]);
+
+                const percentileAllocation: string[] = [];
+                for (const percentile of sortedPercentileList) {
+                    variantsForExperiementation.push(percentile[VARIANT_KEY_NAME]);
+                    percentileAllocation.push(`${percentile[FROM_KEY_NAME]},${base64Helper(percentile[VARIANT_KEY_NAME])},${percentile[TO_KEY_NAME]}`);
+                }
+                rawAllocationId += percentileAllocation.join(";");
+            }
+        }
+
+        if (featureFlag[ALLOCATION_KEY_NAME] === undefined || (variantsForExperiementation.length === 0 && featureFlag[ALLOCATION_KEY_NAME][SEED_KEY_NAME] === undefined)) {

```

Comment:
```suggestion
        if (variantsForExperiementation.length === 0 && featureFlag[ALLOCATION_KEY_NAME][SEED_KEY_NAME] === undefined) {
```

We're doing this same check multiple times so I'd rather combine them. Added a suggestion earlier to do so ^

Reply:
If `featureFlag[ALLOCATION_KEY_NAME]` is undefined, `featureFlag[ALLOCATION_KEY_NAME][SEED_KEY_NAME]` will throw exception.